pkg/costmodel/costmodel.go

```diff
@@ -161,9 +160,8 @@ const (
 	queryCPURequestsStr = `avg(
 		label_replace(
 			label_replace(
-				avg(
-					sum_over_time(kube_pod_container_resource_requests{resource="cpu", unit="core", container!="",container!="POD", node!="", %s}[%s] %s)
-				) by (namespace,container,pod,node,%s) , "container_name","$1","container","(.+)"
+				sum_over_time(kube_pod_container_resource_requests{resource="cpu", unit="core", container!="",container!="POD", node!="", %s}[%s] %s)
```

Comment:
Same as above

Reply:
It's been intentional.

If you take a closer look at the queries.

```
avg(
  label_replace(
     avg (metric) by pod_name,
     pod_name, pod
  )
) by pod
```

Basically, it performs exactly the same aggregation twice.

Reply:
Ok, good catch thank you. @thomasvn since the core queries are changing would love if we took a close look on this in integration tests.

Reply:
@AjayTripathy Yep! Our nightly integration test which validates a pod's allocation costs would catch any issues here.